ffi/diplomat/src/casemap.rs

```diff
@@ -204,4 +184,65 @@ pub mod ffi {
             self.0.simple_fold_turkic(ch)
         }
     }
+
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::casemap::CaseMapCloser, Struct)]
+    pub struct ICU4XCaseMapCloser(pub CaseMapCloser<CaseMapper>);
+
+    impl ICU4XCaseMapCloser {
+        /// Construct a new ICU4XCaseMapper instance
+        #[diplomat::rust_link(icu::casemap::CaseMapCloser::new, FnInStruct)]
+        pub fn create(provider: &ICU4XDataProvider) -> Result<Box<ICU4XCaseMapCloser>, ICU4XError> {
+            Ok(Box::new(ICU4XCaseMapCloser(call_constructor!(
+                CaseMapCloser::new [r => Ok(r)],
+                CaseMapCloser::try_new_with_any_provider,
+                CaseMapCloser::try_new_with_buffer_provider,
+                provider,
+            )?)))
+        }
+
+        /// Adds all simple case mappings and the full case folding for `c` to `builder`.
+        /// Also adds special case closure mappings.
+        ///
+        /// In other words, this adds all characters that this casemaps to, as
+        /// well as all characters that may casemap to this one.
+        ///
+        /// Note that since ICU4XCodePointSetBuilder does not contain strings, this will
+        /// ignore string mappings
+        ///
+        /// Identical to the similarly named method on `ICU4XCaseMapCloser`, use that if you
+        /// do not plan on using string case closure mappings to limit the amount of data loaded.
+        #[cfg(feature = "icu_properties")]
+        #[diplomat::rust_link(icu::casemap::CaseMapCloser::add_case_closure, FnInStruct)]
+        pub fn add_case_closure(
+            &self,
+            c: char,
+            builder: &mut crate::collections_sets::ffi::ICU4XCodePointSetBuilder,
+        ) {
+            self.0.add_case_closure(c, &mut builder.0)
+        }
+
+        /// Maps the string to single code points and adds the associated case closure
+        /// mappings, if they exist.
+        ///
+        /// The string is mapped to code points if it is their full case folding string.
+        /// In other words, this performs a reverse full case folding and then
+        /// adds the case closure items of the resulting code points.
+        /// If the string is found and its closure applied, then
+        /// the string itself is added as well as part of its code points' closure.
+        ///
+        /// Returns true if the string was found
```

Comment:
Nit: Maybe try to keep FFI docs and Rust docs closer; usually what I do is make a one-line summary on the FFI docs and rely on rust_link to give me the full docs